test/watch/test/stdin.test.ts

```diff
@@ -21,63 +21,61 @@ afterEach(() => {
 if (process.env.GITHUB_ACTIONS)
   test.only('skip tests on CI', () => {})
 
-test('quit watch mode', async () => {
-  const vitest = await runVitestCli(...cliArgs)
+describe.each(['--standalone', '--no-standalone'])('watch mode %s', (standalone) => {
+  const allCliArgs = [...cliArgs, standalone]
 
-  vitest.write('q')
+  test('quit watch mode', async () => {
+    const vitest = await runVitestCli(...allCliArgs)
 
-  await vitest.isDone
-})
+    vitest.write('q')
 
-test('rerun current pattern tests', async () => {
-  const vitest = await runVitestCli(...cliArgs, '-t', 'sum')
+    await vitest.isDone
+  })
 
-  vitest.write('r')
+  test('filter by filename', async () => {
+    const vitest = await runVitestCli(...allCliArgs)
 
-  await vitest.waitForStdout('Test name pattern: /sum/')
-  await vitest.waitForStdout('1 passed')
-})
+    vitest.write('p')
 
-test('filter by filename', async () => {
-  const vitest = await runVitestCli(...cliArgs)
+    await vitest.waitForStdout('Input filename pattern')
 
-  vitest.write('p')
+    vitest.write('math')
 
-  await vitest.waitForStdout('Input filename pattern')
+    await vitest.waitForStdout('Pattern matches 1 result')
+    await vitest.waitForStdout('› math.test.ts')
 
-  vitest.write('math')
+    vitest.write('\n')
 
-  await vitest.waitForStdout('Pattern matches 1 result')
-  await vitest.waitForStdout('› math.test.ts')
+    await vitest.waitForStdout('Filename pattern: math')
+    await vitest.waitForStdout('1 passed')
+  })
 
-  vitest.write('\n')
+  test('filter by test name', async () => {
+    const vitest = await runVitestCli(...allCliArgs)
 
-  await vitest.waitForStdout('Filename pattern: math')
-  await vitest.waitForStdout('1 passed')
-})
+    vitest.write('t')
 
-test('filter by test name', async () => {
-  const vitest = await runVitestCli(...cliArgs)
+    await vitest.waitForStdout('Input test name pattern')
 
-  vitest.write('t')
+    vitest.write('sum')
+    if (standalone === '--standalone')
+      await vitest.waitForStdout('Pattern matches no results')
```

Comment:
It's not possible to start in standalone mode and type test name pattern?

Reply:
It's possible to type it, but it's not possible to select it. The selection is provided by the `vitest.state` which is empty since no tests are collected at the start of a standalone mode.